Source/WebKit/UIProcess/Cocoa/SOAuthorization/SOAuthorizationSession.mm

```diff
@@ -87,12 +87,13 @@ static bool isSameOrigin(const WebCore::ResourceRequest& request, const WebCore:
 
 } // namespace
 
-SOAuthorizationSession::SOAuthorizationSession(Ref<API::NavigationAction>&& navigationAction, WebPageProxy& page, InitiatingAction action)
+SOAuthorizationSession::SOAuthorizationSession(RetainPtr<WKSOAuthorizationDelegate> delegate, Ref<API::NavigationAction>&& navigationAction, WebPageProxy& page, InitiatingAction action)
     : m_soAuthorization(adoptNS([PAL::allocSOAuthorizationInstance() init]))
     , m_navigationAction(WTFMove(navigationAction))
     , m_page(page)
     , m_action(action)
 {
+    m_soAuthorization.get().delegate = delegate.get();
```

Comment:
Does simple assignment not work?

m_soAuthorization = delegate?

Reply:
We are assigning to the delegate inside of `m_soAuthorization`. I think `m_soAuthorization.delegate = delegate.get()` will complain `RetainPtr<...>` doesn't have a delegate.